packages/components/src/Icon/Icon.test.js

```diff
@@ -37,4 +50,50 @@ describe('Icon', () => {
 		const wrapper = shallow(<Icon name="svg-dd" className="custom-class" data-custom="hello" />);
 		expect(wrapper.getElement()).toMatchSnapshot();
 	});
+	it('should support remote svg', async () => {
+		const cache = {};
+		const mockIconResponseText = new Promise(resolve => {
+			cache.resolve = () => {
+				resolve('<svg><g><path d=""/></g></svg>');
+				return Promise.resolve({});
+			};
+		});
+		const mockFetchPromise = new Promise(resolve => {
+			cache.first = () => {
+				resolve({
+					status: 200,
+					ok: true,
+					headers: {
+						get: () => 'image/svg+xml',
+					},
+					text: () => mockIconResponseText,
+				});
+				return Promise.resolve({});
+			};
+		});
+		jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise);
+		// test first render
+		const wrapper = mount(<Icon name="remote-/assets/icons/my-icon.svg" />);
+		expect(wrapper.getElement()).toMatchSnapshot();
+
+		await act(async () => {
+			await cache.first();
+			await cache.resolve();
+			wrapper.update();
+		});
+
+		// expect(container.querySelector('svg')).toMatchSnapshot();
+		// act(() => {
+		// 	setTimeout(() => {}, 1);
+		// });
+		// expect(container.querySelector('svg')).toMatchSnapshot();
+		expect(global.fetch).toHaveBeenCalledTimes(1);
+		expect(global.fetch).toHaveBeenCalledWith('/assets/icons/my-icon.svg', {
+			headers: { Accept: 'image/svg+xml' },
+		});
+		const instance = wrapper.children().getElement();
+		const ref = {};
+		instance.ref(ref);
+		expect(ref.innerHTML).toMatchSnapshot();
```

Comment:
I didn t find any other way to make the ref callback called ... I am opened to any other option

Reply:
The idea is to get the resulting html, why don't we use [Enzyme .html()](https://enzymejs.github.io/enzyme/docs/api/ShallowWrapper/html.html).

Something like `wrapper.find('svg').html()`